controllers/rabbitmqcluster_controller.go

```diff
@@ -113,19 +113,12 @@ func (r *RabbitmqClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 		}
 	}
 
-	childResources, err := r.getChildResources(ctx, *rabbitmqCluster)
+	requeueAfter, err := r.updateStatus(ctx, rabbitmqCluster)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
```

Comment:
Perhaps we can merge the `if` conditions as `if err != nil || requeueAfter > 0` and always return with `RequeueAfter` ?

In the case of `err = nill` and `requeueAfter = 2`, we will requeue without printin the stack trace; otherwise we will requeue after 0 and print the error. I believe it is the same behaviour.

Reply:
Oh yes good spot, thanks @Zerpet :)

Looking at https://github.com/kubernetes-sigs/controller-runtime/blob/2b423ec0a6646ad7fec022f76aa7c94358dc1135/pkg/internal/controller/controller.go#L261-L282, merging both branches / conditions has the same behaviour.